examples/pytorch/image-classification/run_image_classification.py

```diff
@@ -268,9 +267,14 @@ def main():
             "imagefolder",
             data_files=data_files,
             cache_dir=model_args.cache_dir,
-            task="image-classification",
         )
 
+    # Rename image and label columns if needed (e.g. Cifar10)
+    if "img" in dataset["train"].features:
```

Comment:
Note that if `data_args.dataset_name` is None and `data_args.train_dir` is None, then the `dataset` dict will not have a `"train"` key, and the line above will raise a `KeyError`.

This could be avoided by replacing the line above with:
```suggestion
    if "img" in list(dataset.column_names.values())[0]:
```

Reply:
Or:
```suggestion
    if "img" in next(iter(dataset.column_names.values())):
```

Reply:
maybe more readable ?
```suggestion
    if "img" in (dataset["train"].features if "train" in dataset else dataset["validation"].features):
```

and also compatible with your suggestion at https://github.com/huggingface/datasets/issues/6571

Reply:
actually it seems this script always do training no ? in this case you can assume "train" is always present

Reply:
Indeed it's probably mostly used for training, but I'm going to add the suggestion anyway in case.